modules/collect/src/test/java/com/opengamma/strata/collect/io/IniFileTest.java

```diff
@@ -174,6 +176,25 @@ public void test_of_ioException() {
         () -> IniFile.of(Files.asCharSource(new File("src/test/resources"), StandardCharsets.UTF_8)));
   }
 
+  @Test
+  public void test_combinedWith() {
+    Map<String, PropertySet> aSections = ImmutableMap.of(
+        "InA", PropertySet.of(ImmutableMap.of("ATest", "AValue")),
+        "InBoth", PropertySet.of(ImmutableMultimap.of("InBoth", "Override", "InBoth", "AlsoOverrides")));
+    IniFile a = IniFile.of(aSections);
+
+    Map<String, PropertySet> bSections = ImmutableMap.of(
+        "InB", PropertySet.of(ImmutableMap.of("BTest", "BValue")),
+        "InBoth", PropertySet.of(ImmutableMultimap.of("InBoth", "Ignored", "InBoth", "AlsoIgnored")));
+    IniFile b = IniFile.of(bSections);
+
+    IniFile combined = a.combinedWith(b);
+    assertThat(combined.sections()).containsExactlyInAnyOrder("InA", "InB", "InBoth");
+    assertThat(combined.section("InA").valueList("ATest")).isEqualTo(ImmutableList.of("AValue"));
+    assertThat(combined.section("InB").valueList("BTest")).isEqualTo(ImmutableList.of("BValue"));
+    assertThat(combined.section("InBoth").valueList("InBoth")).isEqualTo(ImmutableList.of("Override", "AlsoOverrides"));
```

Comment:
This looks like a bug in `PropertySet`. All the keys should be retained on merge AFAICT. The bug is that it uses `valueList()` without considering the need to merge lists.

Reply:
Scratch that, when merging a single key it must be a complete override, so there is no way to accurately merge a list.